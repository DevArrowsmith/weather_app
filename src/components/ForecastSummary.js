import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";

const ForecastSummary = ({
    date,
    temperature,
    description,
    icon,
    onSelect,
}) => {
    return (
        <div className="forecast-summary" data-testid="forecast-summary">
            <div className="forecast-summary__date">
                {moment(date).format("ddd Do MMM")}
            </div>
            <div className="forecast-summary__description">{description}</div>
            <div className="forecast-summary__temperature">
                {temperature.max}Â°C
            </div>
            <div className="forecast-summary__icon" data-testid="forecast-icon">
                <WeatherIcon name="owm" iconId={icon.toString()} />
            </div>
            <button type="button" onClick={() => onSelect(date)}>
                More details
            </button>
        </div>
    );
};

ForecastSummary.propTypes = {
    date: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
    temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
    }).isRequired,
};

export default ForecastSummary;
