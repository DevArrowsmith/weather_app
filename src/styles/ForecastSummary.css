.forecast-summary {
    margin: 0 20px;
}

.forecast-summary__date {
    margin-bottom: 10px;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    font-weight: bold;
}

.forecast-summary__icon, .forecast-summary__description, .forecast-summary__temperature {
    margin: 10px 0;
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: center;
}

.forecast-summary__button {
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: center;
}
