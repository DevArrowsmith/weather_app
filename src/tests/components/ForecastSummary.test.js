import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
    const validProps = {
        date: 111111,
        description: "Stub description",
        icon: "800",
        temperature: {
            min: 12,
            max: 34,
        },
    };

    it("renders correctly (snapshot)", () => {
        const { asFragment } = render(
            <ForecastSummary
                date={validProps.date}
                description={validProps.description}
                icon={validProps.icon}
                temperature={validProps.temperature}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders using the correct props", () => {
        const { getByText, getByTestId } = render(
            <ForecastSummary
                date={validProps.date}
                description={validProps.description}
                icon={validProps.icon}
                temperature={validProps.temperature}
            />
        );

        expect(getByText("111111")).toHaveClass("forecast-summary__date");
        expect(getByText("Stub description")).toHaveClass(
            "forecast-summary__description"
        );
        expect(getByTestId("forecast-icon")).toHaveClass(
            "forecast-summary__icon"
        );
        expect(getByText("34Â°C")).toHaveClass("forecast-summary__temperature");
    });
});
