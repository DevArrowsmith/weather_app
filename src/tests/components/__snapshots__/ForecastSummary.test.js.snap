// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <h2>
      Forecast Summary
    </h2>
    <div
      class="forecast-summary__date"
    >
      111111
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary__temperature"
    >
      34Â°C
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      Stub icon
    </div>
  </div>
</DocumentFragment>
`;
